### PR TITLE
Updating OpenShift glossary items for new guidance

### PR DESCRIPTION
**Pull request by bergerhoffer**

### CODE REVIEW

**Comment by bergerhoffer** [comment]
The OpenShift docs team recently updated guidance on how some terms are referenced. For example, all instances of Pod used to be capitalized, but now pod is lowercase by default. Only references to actual API objects are capitalized (and monospaced as literal values).

Updating the SSG glossary per these updates. A wider FYI on these changes is expected to go out shortly.

Additional info can be found in OpenShift's guidelines here: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 1]
```diff
@@ -101,6 +107,17 @@ definition and a set of triggers for when a new build should be created.
...
 See xref:cluster[cluster] in the _General Conventions_ part.
 
+[discrete]
+[[config-map]]
+==== config map (noun)
+*Description*: A config map holds configuration data for pods to consume. The API object for a config map is `ConfigMap`.
+
+*Use it*: yes
+
+*Incorrect forms*:
```
Comment (adellape) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:117]: List plain-text "configmap" and "configuration map" here?

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 2]
```diff
@@ -117,9 +134,9 @@ See xref:container[container] in the _General Conventions_ part.
 *See also*:
 
 [discrete]
-[[deployment]]
-==== deployment (noun)
-*Description*: A statement of intent by a user to roll out a new version of a configuration. A replication controller can be used for that and other purposes. All deployments are replication controllers, but not all replication controllers are deployments. To avoid confusion, do not refer to an overall OpenShift installation, instance, or cluster as an "OpenShift deployment".
+[[custom-resource]]
+==== custom resource (noun)
+*Description*: A custom resource (CR) is a resource implemented through the Kubernetes `CustomResourceDefinition` API. A custom resource is distinct from the built-in Kubernetes resources, such as the pod and service resources. Every CR is part of an API group.
```
Comment (adellape) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:139]: I wonder if "distinct from" is too strong, making it sound like they behave differently or are somehow managed more specially than the built-in resources. Maybe to add a bit more context as to why they're distinct, if sticking with that wording:

"While custom resources behave the same as the built-in set of Kubernetes and OpenShift resources, such as the pod and service resources, custom resources are distinct in that they are added to the cluster by administrators either manually or by installing Operators, and therefore may not be available on all clusters by default.

That's a long sentence but you get the idea. Also added "OpenShift resources" in there so that one doesn't assume that we consider the default OpenShift-provided ones as "custom resources" either (I'm not actually sure whether technically they are, though).

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 3]
```diff
@@ -128,31 +145,28 @@ See xref:container[container] in the _General Conventions_ part.
...
-*See also*: xref:api-objects[API objects]
+*See also*:
 
 [discrete]
-[[deployment-controller]]
-==== deployment controller (noun)
-*Description*: A Kubernetes object that creates a replication controller from a given Pod template. If that Pod template is modified, the deployment controller creates a new replication controller based on the modified Pod template and replaces the old replication controller with this new one.
+[[deployment]]
+==== deployment (noun)
+*Description*: A statement of intent by a user to roll out a new version of a configuration. A replication controller can be used for that and other purposes. All deployments are replication controllers, but not all replication controllers are deployments. To avoid confusion, do not refer to an overall OpenShift installation, instance, or cluster as an "OpenShift deployment".
```
Comment (adellape) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:161]: "A replication controller can be used for that and other purposes. All deployments are replication controllers, but not all replication controllers are deployments."

These (old) sentences are strange/vague (first sentence) or not worth mentioning (second sentence) to me, but also leaves out replica sets, which at this point are more common than replication controllers AFAIK (since Deployments are preferred to DeploymentConfigs). I might just remove them, and possibly also add a sentence that asscoiates Deployments with ReplicaSets and DeploymentConfigs with RelicationControllers (I notice now that "replication controller" has an entry but "replica set" does not).

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 4]
```diff
@@ -128,31 +145,28 @@ See xref:container[container] in the _General Conventions_ part.
...
+
+The API object for a deployment can be either a Kubernetes-native `Deployment` object or an OpenShift-specific `DeploymentConfig` object.
 
 *Use it*: yes
 
 *Incorrect forms*:
```
Comment (adellape) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:167]: Add "deployment configuration" here?

**Review by adellape — APPROVED** [review]
LGTM overall. Some musings inline.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc (hunk 3 above)
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:161]: Thanks. Agree - I've removed those two sentences. And below where I mention the two deployment objects, I added whether they use replication controllers or replica sets.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc (hunk 2 above)
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:139]: Thanks, I've replaced that sentence with your suggestion.

**Comment by bergerhoffer** [comment]
@redhat-documentation/ccs-style-council For review please. I've asked another writer on the OpenShift team (@adellape) to go through and confirm the updates.

**Review by adellape — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@stoobie @ndeevy @cbudz @IngridT1 @inoxx03 @jhoyt-rh @JoanneOFlynn2018 @gregraka @leswilliams44 

Sorry for the hurry on this, but can I get some reviews/approvals (if appropriate) on this? I need to get an email out this week to announce these changes and would like to have these updates in before that.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 5]
```diff
@@ -18,17 +18,23 @@
 *Description*: API objects are the resources that can be exposed by an API server
 at an endpoint for interrogation.
 
-Use initial capitalization and camel case for Kubernetes or OpenShift API
-objects and do not mark them up unless referring to a specific field or variable
-name from a spec or manifest file.
+When referring to terms that are API objects, if it is important to make the distinction that the context is in reference to the actual API object (like when defining or editing an object), use the proper PascalCase `Kind` for the object, and mark it up in monospace. For example:
```
Comment (cbudz) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:21]: I would suggest revising this initial sentence to be more directly prescriptive: "When referring to an API object that shares its name with a common term, use the proper PascalCase `Kind` for the object, and mark it up in monospace. For example:"

**Review by cbudz — CHANGES_REQUESTED** [review]
Hey @bergerhoffer left one comment. other than that looks good to me.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc [hunk 6]
```diff
@@ -18,17 +18,23 @@
...
+* `DaemonSet`
+* `ConfigMap`
 
-- Pod
-- Deployment
-- Operator
-- DaemonSet (and not "daemon set", "daemonset", or "Daemonset")
+Example object-specific usage: _The default amount of CPU that a container can use if not specified in the `Pod` spec._
+
+However, any other general references to the terms should use lower case, and separate multi-word API objects. For example:
```
Comment (ndeevy) [inline supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc:30]:
```suggestion
However, use lowercase for any other general references to the terms, and separate multi-word API objects. For example:
```

IBM style suggestions here: remove the 'should', the passive, and change spelling of lowercase

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc (hunk 5 above)
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc]: Thanks! I've reordered (because we want people using the general reference guidance if possible) and clarified a bit so it's not so long winded.

File: supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc (hunk 6 above)
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/product_conventions/openshift.adoc]: Thanks, removed!

**Comment by bergerhoffer** [comment]
@cbudz @ndeevy I reordered the two options and tried to simplify the wording and get rid of "should". Can you take another look?

**Review by cbudz — APPROVED** [review]

**Comment by ndeevy** [comment]
Overall it looks good @bergerhoffer and because there's a rush I think it's fine. Since it's merged I won't comment (sorry had a mtg interrupt my review :-) Just to mention for future fixes though, the entry still breaks a good few IBM style rules (passives, phrasal verbs, anthropomorphism etc.) -nothing more serious though! Thanks for doing this and sorry for my delay.

**Comment by bergerhoffer** [comment]
@ndeevy Yes sorry about the rush to merge, but I'm definitely happy to open a new PR to address any issues you found, if you want to comment with what you've noticed

**Comment by ndeevy** [comment]
> @ndeevy Yes sorry about the rush to merge, but I'm definitely happy to open a new PR to address any issues you found, if you want to comment with what you've noticed

Ok, can do. Or if you'd prefer me to open a new PR with suggested changes I can do that too.

**Comment by bergerhoffer** [comment]
> > @ndeevy Yes sorry about the rush to merge, but I'm definitely happy to open a new PR to address any issues you found, if you want to comment with what you've noticed
> 
> Ok, can do. Or if you'd prefer me to open a new PR with suggested changes I can do that too.

@ndeevy Whatever is easiest for you, just let me know!

**Comment by ndeevy** [comment]
> > > @ndeevy Yes sorry about the rush to merge, but I'm definitely happy to open a new PR to address any issues you found, if you want to comment with what you've noticed
> > 
> > 
> > Ok, can do. Or if you'd prefer me to open a new PR with suggested changes I can do that too.
> 
> @ndeevy Whatever is easiest for you, just let me know!

@bergerhoffer OK, I'll open a PR :)
